client/verta/verta/_internal_utils/_config_utils.py

```diff
@@ -35,25 +48,46 @@ def write_config():
     """
     Updates the nearest config file.
 
+    Yields
+    ------
+    config : dict
+        Contents of the nearest config file.
+
     """
-    # TODO: unify with Client's config methods
-    # TODO: recursively search for nearest config file
-    config_filepath = CONFIG_YAML_FILENAME
+    config_filepath = find_closest_config_file()
```

Comment:
If the closest one is in the home folder, then you'll end up setting a global config even for local configs. I'd recommend keeping it local for almost all configs (I don't recall which configs we have).

Also, it seems like it will dump the whole merged config, right? This seems wrong, as now you can't change the global config (e.g. in home).

If this is too complicated to handle, then we can live with a single config for now.

Reply:
1. That's a good point. How about if the nearest config is the home config, then instead create and use a new empty config in the current directory? I think it makes sense for all the items (listed [here](https://github.com/VertaAI/modeldb/blob/master/protos/protos/public/client/Config.proto))
1. This util will _not_ dump the whole merged config into the nearest config! This only gives the contents of the one nearest config file to edit; the full merged config won't be copied into it. The flow with these utils is:
    1. To get the whole merged config for reading only, use the other `read_config` util.
    1. To make config changes, close the `read_config`, stop using the merged `dict` from that util, use _this_ `write_config` util which _only_ exposes the nearest config, and edit/insert its values.

Reply:
(1) sgtm. (2) sounds very complicated. This isn't exposed to the user, right? I'd call it `write_local_config` to clarify.

Reply:
2) It's not exposed to the user; this is only for the Client to use internally. `write_local_config()` is a good idea! I think I'd rename the read util to `read_merged_config()` as well.

Reply:
1. https://github.com/VertaAI/modeldb/pull/635/commits/0a44b1130a8094051fcda3bacb8a7bf54a70c3a1
2. https://github.com/VertaAI/modeldb/pull/635/commits/53f509df0dead0db19678dfb0499a08f7aa544fd